content/lectures/lec00.md

```diff
@@ -7,11 +7,203 @@ slides: https://docs.google.com/presentation/d/1tUFsV_Df2ZB_N8zjG-w9coywL_mYAOGm
+
+### Python
+
+We'll be using Python for most of the development side of the DevOps solutions we cover in this course. It's common and well supported in the infrastructure space because it's easy to learn and there is wide library support.
+
+#### Python example
+```python
+# Comments start with a `#`
+
+import time # import a module from the standard library
+
+for i in range(1, 16): # For loops only over iterators like lists and `range()`
+    print(i)
+    if i % 3 == 0 and i % 5 == 0:  # Conditional expressions
+        print(time.time())
+        print("fizzbuzz")  # Strings can be double or single quoted
+    elif i % 3 == 0:
+        print("fizz")
+    elif i % 5 == 0:
+        print("buzz")
+```
+
+Running the above code produces the following output:
+```bash
+$ python3 test.py
+1
+2
+3
+fizz
+4
+5
+buzz
+6
+fizz
+7
+8
+9
+fizz
+10
+buzz
+11
+12
+fizz
+13
+14
+15
+1607229328.9530184
+fizzbuzz
+```
+
+Code in Python can run at the top-level, but it's good practice to pull logic into functions:
+
+```python
+def get_buzz(i): # (def)ine a function
+    if i % 3 == 0 and i % 5 == 0:
+        return "fizzbuzz"
+    elif i % 3 == 0:
+        return "fizz"
+    elif i % 5 == 0:
+        return "buzz"
+    return ""
+
+for i in range(1, 16):
+    print(str(i) + " " + get_buzz(i)) # Call a function
+
+```
+
+You can check out [CIS 192](cis192.github.io) for more learning materials, and come to office hours with any questions about Python!
+
+### Packaging
+
+Writing code is useful, reusing code is even more useful! Making sure that you has access to the right packages and are also using the correct *version* of that package is no easy task. Python's default package manager, `pip`, will simply take the most recent version of a package and (and its dependencies) and pull it down, not taking into account compatibility with other dependencies and potential conflicts a mismatch might cause. Other package managers built on top of pip, like Poetry, help solve this problem. Poetry is like NPM for JavaScript, or Maven for Java. Take a look that the [Poetry demo](#poetry) below for a more in-depth explanation!
+
+Most importantly, Poetry helps us create **reproduceable build environments** wherever we run our code: on our local machines, on our friends's machines, or even on a production server somewher in "the cloud."
+
+#### How it works
+
+Poetry creates and manages two files. `pyproject.toml` is Poetry's **dependency file**: a human-readable (and writeable) file which declares "acceptable versions" of packages, generally a range, such as "1.1 - 1.12", if version 2.0 contains a breaking change. `poetry.lock` is an autogenerated file used to declare *specific* package versions, including dependencies of dependencies. Poetry uses the lock file to save and persist its resolution of conflicts that it resolves from the list in `pyproject.toml`.
```

Comment:
define lockfile in the same format you define dependency file, with the bolding and definition